### ISSUE TITLE
Support DAP - Debug Adapter Protocol

### ISSUE THREAD

**Issue by XVilka** [radare/radare2#12128]
While radare2 is low-level debugger, sometimes it is being used with some source-level debugging too.
Microsoft started an initiative very similar to LSP (Language Server Protocol) but targeting debuggers.
It is called DAP - Debug Adapter Protocol. In this issue I encourage you to think if this kind of support may benefit radare2 or not.

See more information at https://microsoft.github.io/debug-adapter-protocol/

Specification is here https://microsoft.github.io/debug-adapter-protocol/specification

See implementation for C/C++ https://github.com/Microsoft/vscode-cpptools/tree/master/Documentation/Debugger

**Comment by zadlg** [comment]
useless I think

**Comment by gogo2464** [comment]
I do not think this feature is useless. This protocol is the only remote debugger protocol handled by a gameboy emulator: Emulicious. This protocol is usefull for retro-console debugging.

I plan to write a paper about gameboy console for r2conf. Have this feature could be a real plus!

**Comment by radare** [comment]
You can look into that page to see different implementations 


https://microsoft.github.io/debug-adapter-protocol//debug-adapter-protocol/implementors/adapters/

**Comment by gogo2464** [comment]
@radare error 404: are you sure that your link is not dead?

**Comment by radare** [comment]
Funny they check for the referer

https://microsoft.github.io/debug-adapter-protocol//debug-adapter-protocol/

**Comment by gogo2464** [comment]
> 
> Funny they check for the referer https://microsoft.github.io/debug-adapter-protocol//debug-adapter-protocol/

They also check for the refere here

**Comment by Kreijstal** [comment]
> Funny they check for the referer
> 
> https://microsoft.github.io/debug-adapter-protocol//debug-adapter-protocol/

link is 404
